### ISSUE TITLE
Bug: @typescript-eslint/no-shadow triggers on enum items

### ISSUE THREAD

**Issue by sergeykimaia** [typescript-eslint/typescript-eslint#5179]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=KYDwDg9gTgLgBMAdgVwLZwKIA0CMcDecAvgFAmiSwIrrYBMBJcmucAvHAOQCGiAnjAAWAS0QBzTiVJA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7ujABtcNhyJo0DtEhSIAXSkBfEOqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
export enum EX1 { }

export enum EX2 {
  EX1 = 'anything'
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-shadow": [
      "error"
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

no error on EX1 declaration in EX2

### Actual Result

error on EX1 declaration

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^5.27.1` |
| `@typescript-eslint/parser`            | `^5.27.1` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16.15.1` |

**Comment by bradzacher** [comment]
The rule is correct.
Enum members declare variables withim the enum scope.

Checkout the rule docs here for more info and why you need to be careful.
https://typescript-eslint.io/rules/prefer-literal-enum-member
